extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/data/input/kafkainput/KafkaInputFormat.java

```diff
@@ -0,0 +1,163 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.kafkainput;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.apache.druid.data.input.InputEntity;
+import org.apache.druid.data.input.InputEntityReader;
+import org.apache.druid.data.input.InputFormat;
+import org.apache.druid.data.input.InputRowSchema;
+import org.apache.druid.data.input.impl.ByteEntity;
+import org.apache.druid.data.input.impl.DimensionsSpec;
+import org.apache.druid.data.input.impl.TimestampSpec;
+import org.apache.druid.data.input.kafka.KafkaRecordEntity;
+import org.apache.druid.java.util.common.DateTimes;
+import org.joda.time.DateTime;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.util.Collections;
+import java.util.Objects;
+
+public class KafkaInputFormat implements InputFormat
+{
+  private static final String DEFAULT_HEADER_LABEL_PREFIX = "kafka.header.";
+  private static final String DEFAULT_TIMESTAMP_LABEL_PREFIX = "kafka.";
+  private static final String DEFAULT_KEY_LABEL_PREFIX = "kafka.";
+
+  private final KafkaHeaderFormat headerReader;
+  private final InputFormat valueFormat;
+  private final InputFormat keyFormat;
+  private final String headerLabelPrefix;
+  private final String keyLabelPrefix;
+  private final String recordTimestampLabelPrefix;
+
+  public KafkaInputFormat(
+      @JsonProperty("headerFormat") KafkaHeaderFormat headerReader,
+      @JsonProperty("keyFormat") InputFormat keyFormat,
+      @JsonProperty("valueFormat") InputFormat valueFormat,
+      @JsonProperty("headerLabelPrefix") @Nullable String headerLabelPrefix,
+      @JsonProperty("keyLabelPrefix") @Nullable String keyLabelPrefix,
+      @JsonProperty("recordTimestampLabelPrefix") @Nullable String recordTimestampLabelPrefix
+  )
+  {
+    this.headerReader = Preconditions.checkNotNull(headerReader, "headerFormat");
+    this.keyFormat = Preconditions.checkNotNull(keyFormat, "keyFormat");
+    this.valueFormat = Preconditions.checkNotNull(valueFormat, "valueFormat");
+    this.headerLabelPrefix = headerLabelPrefix != null ? headerLabelPrefix : DEFAULT_HEADER_LABEL_PREFIX;
+    this.keyLabelPrefix = keyLabelPrefix != null ? keyLabelPrefix : DEFAULT_KEY_LABEL_PREFIX;
+    this.recordTimestampLabelPrefix = recordTimestampLabelPrefix != null ? recordTimestampLabelPrefix : DEFAULT_TIMESTAMP_LABEL_PREFIX;
+  }
+
+  @Override
+  public boolean isSplittable()
+  {
+    return false;
+  }
+
+  @Override
+  public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
+  {
+    KafkaRecordEntity record = (KafkaRecordEntity) source;
+    TimestampSpec dummyTimestampSpec = new TimestampSpec("__kif_auto_timestamp", "auto", DateTimes.of("2021-06-24"));
```

Comment:
can we add a comment to explain why we need a dummy timestamp spec?
Also for the timestamp this seems a bit arbitrary, maybe we can use epoch 0 to avoid surprises